odoo/orm/fields_properties.py

```diff
@@ -29,6 +29,74 @@ def check_property_field_value_name(property_name):
     if not (0 < len(property_name) <= 512) or not regex_alphanumeric.match(property_name):
         raise ValueError(f"Wrong property field value name {property_name!r}.")
 
+class Property(abc.Mapping):
+    """An object that implements the value of a :class:`Properties` field in the "record"
+    format, i.e., the result of evaluating an expression like ``record.property_field``.
+    The value behaves as a ``dict``, and individual properties are returned in their
+    expected type, according to ORM conventions.  For instance, the value of a many2one
+    property is returned as a recordset::
+
+        # attributes is a properties field, and 'partner_id' is a many2one property;
+        # partner is thus a recordset
+        partner = record.attributes['partner_id']
+        partner.name
+
+    When the accessed key does not exist, i.e., there is no corresponding property
+    definition for that record, the access raises a :class:`KeyError`.
+    """
+
+    def __init__(self, values, field, record):
+        self._values = values or {}
+        self.record = record
+        self.field = field
+
+    def __len__(self):
+        return len(self._values)
+
+    def __iter__(self):
+        return iter(self._values)
+
+    def __getitem__(self, property_name):
+        """Will make the verification."""
+        if not self.record:
+            return False
+
+        records = self.record | self.record.browse(islice(self.record._prefetch_ids, 1000))
+        records = records.filtered(lambda r: r[self.field.definition_record] == self.record[self.field.definition_record])
+
+        values = self.field.convert_to_read(records.env.cache.get(records, self.field), self.record, use_display_name=False)
```

Comment:
I have the feeling that the overhead of `convert_to_read` is stupidly high.  Instead of mapping the cached value to the "read" format directly, you should map it to the "record" format, and then to the "read" format.

Moreover, method `cache.get()` takes one record, and you pass it all the records, which is obviously wrong.  If you fix this, you will realize that `records` is useless (it was already, but it will no longer be used.)

Reply:
> I have the feeling that the overhead of convert_to_read is stupidly high. Instead of mapping the cached value to the "read" format directly, you should map it to the "record" format, and then to the "read" format.

How do you batch the check in `_get_res_ids_per_model` ?

Reply:
> I have the feeling that the overhead of convert_to_read is stupidly high. Instead of mapping the cached value to the "read" format directly, you should map it to the "record" format, and then to the "read" format.

How do you batch `_get_res_ids_per_model` ?

Ok to remove `recrods`

Reply:
Or we need `convert_to_record_multi`

Reply:
No, introducing prefetching is much more subtle than that.  You will need something like https://github.com/odoo/odoo/blob/61d44b354c9dc511ffe79635b591f18b572e2fdc/odoo/fields.py#L5334.